### PR TITLE
agent subscription to multiple groups (orb-tags)

### PR DESCRIPTION
**Pull request by manrodrigues**
- I changed the logic from groups and agents creation in order to enable multiple and defined tags
- Scenarios are:
       - subscribe an agent to multiple groups created before agent provision
       - subscribe an agent to multiple groups created after agent provision
      - create agent without tags
      - create agent with multiple tags
      - create agent group without tags
      - create agent group with multiple tags

### CODE REVIEW

File: python-test/features/agentProviderUi.feature [hunk 1]
```diff
@@ -5,14 +5,14 @@ Feature: Create agents using orb ui
...
-        When a new agent is created through the UI
+        When a new agent is created through the UI with defined orb tag(s): region:br, demo:true, ns1:true
         Then the agents list and the agents view should display agent's status as New within 10 seconds
 
     Scenario: Provision agent
         Given the Orb user logs in through the UI
             And that fleet Management is clickable on ORB Menu
             And that Agents is clickable on ORB Menu
-        When a new agent is created through the UI
+        When a new agent is created through the UI with random orb tag(s): 2 tag(s)
```
Comment (leoparente) [inline python-test/features/agentProviderUi.feature:15]: don't get the "2 tags()". what that does mean?

File: python-test/features/steps/page_objects.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#XPATHs
```
Comment (leoparente) [inline python-test/features/steps/page_objects.py:1]: this was a good idea. maybe add a namespace to the variables so we can know that it comes from this file.

like a prefix to know that the variables are coming from this file.  ex: orb_page_*

Or add a class and export variables through that. example OrbPage object and you get `OrbPage.fleet_management_left_menu` and so on.

I think that second approach is more Object Oriented.

**Review by leoparente — CHANGES_REQUESTED** [review]

File: python-test/features/agentProviderUi.feature (hunk 1 above)
Comment (manrodrigues) [reply python-test/features/agentProviderUi.feature:15]: it means the amount of tags that must be added to the agent or to the group..

File: python-test/features/steps/page_objects.py (hunk 2 above)
Comment (manrodrigues) [reply python-test/features/steps/page_objects.py:1]: oh, interesting and good idea.. i'm gonna working on that

**Review by leoparente — DISMISSED** [review]

**Review by leoparente — APPROVED** [review]

**Pull request merged**